src/main/java/com/patchworkmc/event/initialization/RegisterEventRegistrars.java

```diff
@@ -1,56 +1,65 @@
 package com.patchworkmc.event.initialization;
 
-import java.util.Map;
+import java.util.Set;
 import java.util.function.Consumer;
 
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
 import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 
-public class RegisterEventRegistrars implements Consumer<MethodVisitor> {
-	private Iterable<Map.Entry<String, String>> staticEventRegistrars;
-	private Iterable<Map.Entry<String, String>> instanceEventRegistrars;
+import com.patchworkmc.Constants.EventBus;
+import com.patchworkmc.Constants.Lambdas;
 
-	public RegisterEventRegistrars(Iterable<Map.Entry<String, String>> staticEventRegistrars, Iterable<Map.Entry<String, String>> instanceEventRegistrars) {
-		this.staticEventRegistrars = staticEventRegistrars;
-		this.instanceEventRegistrars = instanceEventRegistrars;
+public class RegisterEventRegistrars implements Consumer<MethodVisitor> {
+	private static final String EVENT_REGISTRAR_REGISTRY = "net/minecraftforge/eventbus/api/EventRegistrarRegistry";
+	private static final String EVENT_REGISTRAR_REGISTRY_INSTANCE_SIGNATURE = "Lnet/minecraftforge/eventbus/api/EventRegistrarRegistry;";
+	private final Set<String> staticClasses;
+	private final Set<String> instanceClasses;
+
+	public RegisterEventRegistrars(Set<String> staticClasses, Set<String> instanceClasses) {
+		this.staticClasses = staticClasses;
+		this.instanceClasses = instanceClasses;
 	}
 
 	@Override
 	public void accept(MethodVisitor method) {
-		for (Map.Entry<String, String> entry : staticEventRegistrars) {
-			String shimName = entry.getKey();
-			String baseName = entry.getValue();
-
-			method.visitFieldInsn(Opcodes.GETSTATIC, "net/minecraftforge/eventbus/api/EventRegistrarRegistry", "INSTANCE", "Lnet/minecraftforge/eventbus/api/EventRegistrarRegistry;");
-
-			method.visitLdcInsn(Type.getObjectType(baseName));
-
-			method.visitTypeInsn(Opcodes.NEW, shimName);
-			method.visitInsn(Opcodes.DUP);
-
-			method.visitMethodInsn(Opcodes.INVOKESPECIAL, shimName, "<init>", "()V", false);
-			method.visitMethodInsn(Opcodes.INVOKEINTERFACE, "net/minecraftforge/eventbus/api/EventRegistrarRegistry", "registerStatic", "(Ljava/lang/Class;Ljava/util/function/Consumer;)V", true);
+		// Making the instance a local variable and pushing it to the stack creates slightly more readable decompiled code.
+		Label start = new Label();
+		method.visitLabel(start);
```

Comment:
What do the labels do here? Just curious.

Reply:
the labels mark the amount of code that is "in scope" for the variable. I haven't tried handing it a `null` one yet but that might give decompilers a heart attack.

Reply:
I can't find a clear explanation on what labels actually *do* functionally, do they affect the JVM or influence decompiler behavior?

Reply:
they're mostly for debugging/compilers, but they provide other things (From the ASM 8 javadocs)

`A position in the bytecode of a method. Labels are used for jump, goto, and switch instructions, and for try catch blocks. A label designates the instruction that is just after. Note however that there can be other elements between a label and the instruction it designates (such as other labels, stack map frames, line numbers, etc.).`

Reply:
Alright, so I guess my main question is whether adding/removing the labels will change the behavior of the method (most likely no) or influence decompiler output (possible?).

Reply:
turns out i wasn't even passing the labels to the localvariable properly and it was still okay, so this visit isn't needed